elastic/kibana
# [Cloud Security] Agentless cloud credentials

## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.
This PR add Automation Credentials buttons for GCP and AWS. UX copy will be addressed in the [folllowing ticket](https://github.com/elastic/kibana/issues/178858).

<img width="900" alt="image" src="https://github.com/elastic/kibana/assets/17135495/f7329fa7-ae48-4dc0-a7cf-a0a26d80ad6b">

For Agentless the CSPM AWS integrations, users  will able to : 
* click Launch Cloud Formation button for single account or organization account  which opens external link to Cloud Formation page.
*  create a cloud formation stack that generates access keys. 
We will use the Cloud Formation  template url from the package info `getTemplateUrlFromPackageInfo` exposed as common utility in fleet.


<img width="884" alt="image" src="https://github.com/elastic/kibana/assets/17135495/d5e5e7d2-21fe-44bc-88e7-908d29817dfb">

For Agentless the CSPM GCP integrations, users  will able to click Launch Cloud Shell button for single accounts or organization account. We get the cloud shell  template url from the package info `getTemplateUrlFromPackageInfo`


## To test PR:

AWS
1. Log in as an admin in the management account of the AWS Organization you want to onboard
2. Click the Launch CloudFormation button below.
3. (Optional) Change the Amazon region in the upper right corner to the region you want to deploy your stack to
4. Tick the checkbox under capabilities in the opened CloudFormation stack review form: I acknowledge that AWS CloudFormation might create IAM resources.
5. Click Create stack.
6. Once stack status is CREATE_COMPLETE then click the Outputs tab
7. Copy Access Key Id and Secret Access Key then paste the credentials below

Use: `eu-west-1` region to create stacks 

GCP
1.Replace <PROJECT_ID> in the following command with your project ID then copy the command
2. Click Launch Google Cloud Shell, then run the command
3. Check `Trust Repo` and click `CONFIRM` button 
5. Paste and run command in the cloud shell terminal
 Cloud Security team   `gcloud config set project test-project-1-396010 && ORG_ID=693506308612 && ./deploy_service_account.sh`
5. Run 'cat KEY_FILE.json' to view the service account key. Copy and paste it in the CSPM GCP integration.

Internal Team Developer  Tips for testing :
-  Look for `cloudshell_git_branch` in the url and change `8.XX` to `main`. 
- To run script, see command [here](https://p.elstc.co/paste/XyJg0wW8#srnavdS0GZtxapNA1ZcNu9TddLpG+zWP-99KWWpIEnr) 
- 
 Any question for GCP account issue reachout to @orouz
